chapter_1/exercise_1_12/copy_io_nl.c

```diff
@@ -1,18 +1,18 @@
-#include <stdio.h>                                                               
-                                                                                 
-int main()                                                                       
-{                                                                                
-        char c;                                                                  
-        char last_char = -1;                                                     
-        while ((c = getchar()) != EOF) {                                         
-                if(c==' ' || c=='\t' ||  c=='\n'){                               
-                        if(last_char!=' ' &&  last_char!='\t' && last_char!='\n'){
-                                putchar('\n');                                   
-                        }                                                        
-                }                                                                
-                else {                                                           
-                        putchar(c);                                              
-                }       
-                last_char=c;
-        }                                                                        
-}        
+#include <stdio.h>
+
+int main()
+{
+  char c;
+  char last_char = -1;
+  while ((c = getchar()) != EOF) {
+    if(c==' ' || c=='\t' ||  c=='\n'){
+      if(last_char!=' ' &&  last_char!='\t' && last_char!='\n'){
```

Comment:
Can you simplify this logic? It's unnecessarily complicated. You could use "short circuits" inverse logic.

Reply:
Thanks for the comments.
Yes I agree the logic is redundant. 

I simplified it a bit. Not sure if this is what you had in mind. It's not obvious to me if we can get rid of the nested if just with smart logic.